src/structure/matrix.rs

```diff
@@ -803,9 +911,17 @@ impl fmt::Display for Matrix {
 impl PartialEq for Matrix {
     fn eq(&self, other: &Matrix) -> bool {
         if self.shape == other.shape {
+            // self.data
+            //     .clone()
+            //     .into_iter()
+            //     .zip(other.data.clone())
+            //     .all(|(x, y)| nearly_eq(x, y))
+            //     && self.row == other.row
+
+            // Note: the sequential version in this case can be replaced by the parallel version?
```

Comment:
The commented part above is the sequential part that we currently have. The below part is the parallelized part. How can we have `impl PartialEq for Matrix` and have both seq and par versions?
If it is not possible, then can we have the parallel version by default??